src/components/Program/ProgramCard.js

```diff
@@ -6,32 +6,53 @@ import Dropdown from 'antd/lib/dropdown';
 import { useDispatch } from 'react-redux';
 import { Link } from 'react-router-dom';
 import { programActions } from '../../state/ducks';
+import { useUserRole } from '../../hooks';
 import { pathUtils } from '../../routes';
 import { MenuItemLink } from '../_common';
 // css
 import '../../styles/ProgramCard.css';
 
 export default function ProgramCard(props) {
-  const { program } = props;
+  const { program, push } = props;
   const dispatch = useDispatch();
 
-  const menu = (
-    <Menu>
-      <MenuItemLink
-        to={pathUtils.makeEditProgramPath(program.programid)}
-        key="edit"
-        extra=""
-      >
-        Edit Program
-      </MenuItemLink>
-      <MenuItemLink
+  const { userIsAdmin } = useUserRole();
+
+  const handleMenuClick = e => {
+    if (e.key === 'edit') {
+      dispatch(programActions.setEdit(program.programid));
+      // push(pathUtils.makeEditProgramPath(program.programId));
```

Comment:
this can be deleted since you made the `Edit` button a `<MenuItemLink to={pathUtils.makeEditProgramPath(...)} />`

Reply:
Deleting this makes my programDelete stop working, I need to refactor either the delete or the edit, i think there should be some talk on the structure of the components, do we want to handle them with MenuItemLink or with Menu.Item? Thanks Chaz

Reply:
@sebass482 I think that's a great point that y'all should totally talk about! I leave that decision fully up to you guys :D